src/xenia/kernel/xam/xam_user.cc

```diff
@@ -65,6 +65,23 @@ X_HRESULT_result_t XamUserGetXUID_entry(dword_t user_index, dword_t type_mask,
 }
 DECLARE_XAM_EXPORT1(XamUserGetXUID, kUserProfiles, kImplemented);
 
+dword_result_t XamUserGetIndexFromXUID_entry(qword_t xuid, dword_t flags,
+                                             pointer_t<uint32_t> index) {
+  for (uint32_t i = 0; i < XUserMaxUserCount; i++) {
+    auto profile = kernel_state()->xam_state()->GetUserProfile(i);
+    if (!profile) {
+      continue;
+    }
+    if (profile->xuid() == xuid) {
+      *index = i;
+      return X_E_SUCCESS;
+    }
+  }
+
+  return X_E_NO_SUCH_USER;
```

Comment:
Instead of this code you can use: ``ProfileManager::GetUserIndexAssignedToProfile``
It returns ``XUserIndexAny`` if specific XUID isn't assigned to any slot and slot id when it is assigned.

So basically implementation will look like this:

```
const uint8_t user_index = profile_manager()->GetUserIndexAssignedToProfile(xuid);
if(user_index == XUserIndexAny) {
    return X_E_NO_SUCH_USER;
}

*index = user_index;
return X_E_SUCCESS;```

also just in case I would check if that index pointer is even valid and return error if it isn't